spring-kafka/src/main/java/org/springframework/kafka/listener/KafkaMessageListenerContainer.java

```diff
@@ -2848,4 +2849,13 @@ public void onSuccess(Object result) {
 
 	}
 
+	@SuppressWarnings("serial")
+	private static class StopAfterFenceException extends KafkaException {
```

Comment:
Don't we bubble this exception to end-user code?
So, they won't be able to handle it properly since it is `private`.
Do I miss anything?

Thanks

Reply:
No; this is thrown on the task executor thread (`run()` dies). We should, however, retain the root cause.